### ISSUE TITLE
Python: Implement missing syntax highlighting (not fixable with tree-sitter)

### ISSUE THREAD

**Issue by JaagupAverin** [zed-industries/zed#21686]
### Describe the feature

Python syntax highlighting is missing the following features that are not addressable using only tree sitter queries. These features should be possible with LSP Semantic Highlighting and is therefore currently blocked by #7450 and #7296.

Missing features:
- [ ] Highlight types/variables correctly even if they don't match python naming conventions;
- [ ] Highlight parameters (`@variable.parameter`) in function arguments and body;

### Environment

Zephyr Preview 0.165.3

### If applicable, add mockups / screenshots to help present your vision of the feature

```python
# Issue #1: Cannot differentiate between types and variables/constants in a
# deterministic way. Currently this is done using naming conventions
# (CamelCase vs snake_case vs UPPER_CASE), but this does not always work.
# For example single-letter names:


def fn(x: type, y: int) -> None:
    pass


class X:
    pass


Y = 42
fn(X, Y)  # X should be classified as a type; Y should be classified as constant;

```

**Comment by notpeter** [comment]
This requires LSP assistance, so closing in favor of:
- https://github.com/zed-industries/zed/issues/7450
